tools/niminst/niminst.nim

```diff
@@ -749,22 +845,107 @@ proc archiveDist(c: var ConfigData) =
   let oldDir = getCurrentDir()
   setCurrentDir(tmpDir)
   try:
+    # Timestamp to set the files to. This make sure that the created archive
+    # is deterministic with regards to creation time.
+    let timestamp =
+      try:
+        parse(c.commitdate, "yyyy-MM-dd", utc()).toTime()
+      except TimeParseError:
+        # If no time is provided, use epoch 0
+        fromUnix(0)
+
+    const
+      AvoidFragments = [
+        DirSep & ".DS_Store", # macOS directory metadata
+        DirSep & "__MACOSX"   # macOS resource forks
+      ]
+        ## Path fragments to avoid packaging
+
+      DefaultPermission = {fpUserRead, fpUserWrite, fpGroupRead, fpOthersRead}
+        ## Permissions applied to files
+      
+      ExecutablePermission = DefaultPermission + {fpUserExec, fpGroupExec, fpOthersExec}
+        ## Permissions applied to executables/directories
+
+    for path in walkDirRec(proj, {pcFile, pcDir}, checkDir = true):
+      if AvoidFragments.allIt(it notin path):
+        # Normalize permissions so that a reasonably stricter/laxer umask won't
+        # cause undeterminism
+        if fpUserExec in getFilePermissions(path):
+          setFilePermissions(path, ExecutablePermission)
+        else:
+          setFilePermissions(path, DefaultPermission)
+
+        # Set the time of every file/directories to be archived to the
+        # deterministic timestamp
+        setLastModificationTime(path, timestamp)
+
+    # Collect all files to be archived
+    var paths = collect(newSeq):
+      for path in walkDirRec(proj, checkDir = true):
+        if AvoidFragments.allIt(it in path):
```

Comment:
Ajay have sequtils, why not use filterait?

Reply:
Zipped this one to the loop above it. It was originally an external loop since I wanted to avoid having directories in the list, which would be recursed by the archivers. The switch in archivers let me disable recursive travel so there is no need to filter directories any more.